### PR TITLE
feat(Resource Access): Duplicate rule

### PR DESCRIPTION
**Pull request by dmyios**
## Description

Implemented Duplicate Resource Access rule functionnality in listing.
[screen-capture (6).webm](https://github.com/centreon/centreon/assets/108675430/579e963e-657c-4af0-aab8-86aea692bfdd)

**Fixes** # MON-24904

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by cg-tw — DISMISSED** [review]

File: centreon/www/front_src/src/ResourceAccessManagement/Actions/Duplicate/DuplicationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import useValidateName from './useValidateName';
+import useDuplicate from './useDuplicate';
+
+const DuplicationForm = (): React.JSX.Element => {
+  const { validationSchema } = useValidateName();
+  const { submit } = useDuplicate();
+
+  return (
+    <Formik
+      initialValues={{ name: '' }}
```
Comment (Thebarda) [inline centreon/www/front_src/src/ResourceAccessManagement/Actions/Duplicate/DuplicationForm.tsx:13]: Can we initialize the name with name of the duplicated rule with _1 appended?

File: centreon/www/front_src/src/ResourceAccessManagement/Actions/api/adapters.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+export const adaptRule = ({
+  contactGroups,
+  contacts,
+  datasetFilters,
+  description,
+  isActivated,
+  name
+}): object => ({
+  contact_groups: map(prop('id'), contactGroups),
+  contacts: map(prop('id'), contacts),
```
Comment (Thebarda) [inline centreon/www/front_src/src/ResourceAccessManagement/Actions/api/adapters.ts:36]: use [pluck](https://ramdajs.com/docs/#pluck) instead

File: centreon/www/front_src/src/ResourceAccessManagement/Listing/Tests/ResourceAccessRulesListing.cypress.spec.tsx [hunk 3]
```diff
@@ -481,3 +490,112 @@ describe('Listing header actions: mass delete', () => {
...
+    cy.findByText(labelEnterNameForDuplicatedRule).should('be.visible');
+    cy.findByLabelText(labelResourceAccessRuleName).should('be.visible');
+    cy.findByText(labelCancel).should('be.visible');
+    cy.findByText(labelDuplicate).should('be.visible');
+
+    cy.makeSnapshot();
+    cy.findByText(labelCancel).click();
+  });
+
+  it('displays required error message when no duplication name was entered', () => {
```
Comment (Thebarda) [inline centreon/www/front_src/src/ResourceAccessManagement/Listing/Tests/ResourceAccessRulesListing.cypress.spec.tsx:534]:
```suggestion
  it('displays required error message when the name remains empty', () => {
```

File: centreon/www/front_src/src/ResourceAccessManagement/Actions/Duplicate/DuplicateConfirmationDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      confirmDisabled={disabled}
+      labelCancel={t(labelCancel)}
+      labelConfirm={t(labelDuplicate)}
+      labelTitle={t(labelEnterNameForDuplicatedRule)}
+      open={isDialogOpen}
+      submitting={isSubmitting}
+      onCancel={onCancel}
+      onConfirm={submitForm}
+    >
+      <TextField
```
Comment (bdauria) [inline centreon/www/front_src/src/ResourceAccessManagement/Actions/Duplicate/DuplicateConfirmationDialog.tsx:60]: can we use the full width ?

File: centreon/www/front_src/src/ResourceAccessManagement/Actions/api/adapters.ts (hunk 2 above)
Comment (dmyios) [reply centreon/www/front_src/src/ResourceAccessManagement/Actions/api/adapters.ts]: Done

File: centreon/www/front_src/src/ResourceAccessManagement/Actions/Duplicate/DuplicateConfirmationDialog.tsx (hunk 4 above)
Comment (dmyios) [reply centreon/www/front_src/src/ResourceAccessManagement/Actions/Duplicate/DuplicateConfirmationDialog.tsx:60]: Done

**Review by bdauria — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**